src/wallet/db.cpp

```diff
@@ -52,6 +53,24 @@ void CheckUniqueFileid(const CDBEnv& env, const std::string& filename, Db& db)
         }
     }
 }
+
+bool LockEnvDirectory(const fs::path& env_path)
+{
+    // Make sure only a single Bitcoin process is using the wallet directory.
+    fs::path lock_file_path = env_path / ".lock";
+    FILE* file = fsbridge::fopen(lock_file_path, "a"); // empty lock file; created if it doesn't exist.
+    if (file) fclose(file);
+
+    try {
+        static boost::interprocess::file_lock lock(lock_file_path.string().c_str());
```

Comment:
Didn't notice this before, and no need for a change now, but just in case #11687 is somehow merged before this PR, static variable here should move to CDBEnv member variable. (A single static variable can't guard different database directories if different database directories may be opened during the lifetime of the process).